packages/pigeon/lib/swift_generator.dart

```diff
@@ -637,11 +667,21 @@ import FlutterMacOS
     });
   }
 
+  void _writeNilOrValue(Indent indent) {
+    indent.format('''
+
+private func nilOrValue<T>(_ value: Any?) -> T? {
+  if value is NSNull { return nil }
+  return (value as Any) as! T?
```

Comment:
Discussed offline - this is indeed a weird Swift behavior that `as Any` is required to make it work. I will take a closer look but no need to block.